tests/test_adr_validator.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from unittest import TestCase
+
+from tests.stubs import get_dummy_data
+
+from boefjes.job_models import NormalizerMeta
+from boefjes.katalogus.local_repository import LocalPluginRepository
+from boefjes.local import LocalNormalizerJobRunner
+
+
+class ADRValidatorNormalizerTest(TestCase):
+    def test_no_findings(self):
+        local_repository = LocalPluginRepository(Path(__file__).parent.parent / "boefjes" / "plugins")
+        runner = LocalNormalizerJobRunner(local_repository)
+        meta = NormalizerMeta.parse_raw(get_dummy_data("adr-validator-normalize.json"))
+
+        raw = """[{"rule": "TEST-01", "passed": true, "message": ""}]"""
+        output = runner.run(meta, bytes(raw, "UTF-8"))
+
+        self.assertEqual(1, len(output.observations))
+
+        observation = output.observations[0]
+        self.assertEqual(2, len(observation.results))
+
+        self.assertEqual("APIDesignRule", observation.results[0].object_type)
+        self.assertEqual("TEST-01", observation.results[0].name)
+        self.assertEqual("APIDesignRuleResult", observation.results[1].object_type)
+        self.assertEqual("", observation.results[1].message)
+        self.assertEqual(True, observation.results[1].passed)
+
+    def test_with_findings(self):
+        local_repository = LocalPluginRepository(Path(__file__).parent.parent / "boefjes" / "plugins")
+        runner = LocalNormalizerJobRunner(local_repository)
+        meta = NormalizerMeta.parse_raw(get_dummy_data("adr-validator-normalize.json"))
+
+        raw = """[
+            {"rule": "TEST-01", "passed": true, "message": ""},
+            {"rule": "TEST-02", "passed": false, "message": "An error"},
+            {"rule": "TEST-02", "passed": true, "message": ""}
+        ]"""
+        output = runner.run(meta, bytes(raw, "UTF-8"))
+
+        self.assertEqual(1, len(output.observations))
+
+        observation = output.observations[0]
+        self.assertEqual(8, len(observation.results))
+
+        self.assertEqual("APIDesignRule", observation.results[2].object_type)
+        self.assertEqual("TEST-02", observation.results[2].name)
+        self.assertEqual("APIDesignRuleResult", observation.results[3].object_type)
+        self.assertEqual("An error", observation.results[3].message)
```

Comment:
If these aren't strict comparisons, then please use `assertIn` instead

Reply:
These are. I'm testing if the message from the validator is correctly set to the object and finding.